api/controllers/auth/logout.js

```diff
@@ -22,7 +23,23 @@ module.exports = function (req, res) {
 
    // passport session logout feature:
    if (req.logout) {
-      req.logout();
+      if (sails.config.cas.enabled) {
+         var cas = new CasStrategy({
+            casURL: sails.config.cas.baseURL,
+            pgtURL: sails.config.cas.pgtURL || sails.config.cas.proxyURL,
+            sslCert: sails.config.cas.sslCert || null,
+            sslKey: sails.config.cas.sslKey || null,
+            sslCA: sails.config.cas.sslCA || null,
+            passReqToCallback: true,
+         });
+         cas.logout(
+            req,
+            res,
+            `${sails.config.cas.baseURL}/logout?service=${req.tenantUrl}`
```

Comment:
The 3rd param for `cas.logout()` is just the return URL. The CasStrategy will add the CAS base URL on its own. Alternatively, you could manually redirect to the full URL and bypass CasStrategy altogether.

Reply:
Thanks Joshua! I have changed accordingly...for now I will try to use the CasStrategy just in case there is something added in the future that would prevent the manual method from working.